mlrun/api/db/sqldb/db.py

```diff
@@ -543,6 +543,12 @@ def _process_artifact_dict_to_store(artifact, key, iter=None):
             )
         if not db_key:
             artifact["spec"]["db_key"] = key
+
+        data_key = artifact.get("metadata", {}).get("key")
+        if (key and data_key) and data_key != key:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"Artifact with conflicting key name - {data_key} while request key : {key}."
```

Comment:
I suggest having a similar error message for both errors (here and in line 542)
and please add the keys to the message similarly above (and with the legacy artifact)
```suggestion
                f"Conflict between requested key ({key}) and key in artifact metadata ({data_key})"
```